src/xmainloop.cpp

```diff
@@ -334,6 +342,18 @@ void XMainLoop::enternotify(XCrossingEvent* ce) {
         } else if (c) {
             focus_client(c, false, true, false);
         }
+        if (!c) {
+            // if it's not a client window, it's maybe a frame
+            FrameDecoration* frameDec = FrameDecoration::withWindow(ce->window);
+            if (frameDec) {
+                auto frame = frameDec->frame();
+                HSWeakAssert(frame);
+                if (frame) {
+                    root_->focusFrame(frame);
+                }
+            } else {
```

Comment:
Is the empty else condition really necessary?

Reply:
I've removed it (I had a printf debugging statement there before).